app/javascript/controllers/autocomplete_select_controller.js

```diff
@@ -9,19 +9,19 @@ export default class extends Controller {
     useClickOutside(this)
   }
 
+  clickOutside(_event) {
+    this.close()
+  }
+
   toggle() {
     if (this.listTarget.classList.contains("hidden")) {
       this.open()
+      useClickOutside(this)
     } else {
       this.close()
     }
   }
 
-  clickOutside(event) {
-    event.preventDefault()
```

Comment:
This prevented any ClickOutside to not do what should be done, which means submitting buttons were useless. This took me some time to realise.